core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java

```diff
@@ -189,16 +191,22 @@ private void configureHttp(ChannelPipeline p, @Nullable ProxiedAddresses proxied
         }
         final ServerHttp1ObjectEncoder responseEncoder = new ServerHttp1ObjectEncoder(
                 p.channel(), H1C, keepAliveHandler,
-                config.isDateHeaderEnabled(), config.isServerHeaderEnabled()
+                config.isDateHeaderEnabled(),
+                config.isServerHeaderEnabled()
         );
         p.addLast(TrafficLoggingHandler.SERVER);
         p.addLast(new Http2PrefaceOrHttpHandler(responseEncoder));
-        p.addLast(new HttpServerHandler(config, gracefulShutdownSupport, responseEncoder,
-                                        H1C, proxiedAddresses));
+        final HttpServerHandler httpServerHandler = new HttpServerHandler(configHolder,
+                                                                          gracefulShutdownSupport,
+                                                                          responseEncoder,
+                                                                          H1C, proxiedAddresses);
+        p.addLast(httpServerHandler);
+        configHolder.addListener(httpServerHandler.configUpdateListener());
     }
 
     private Timer newKeepAliveTimer(SessionProtocol protocol) {
-        return MoreMeters.newTimer(config.meterRegistry(), "armeria.server.connections.lifespan",
+        return MoreMeters.newTimer(config.meterRegistry(),
+                                   "armeria.server.connections.lifespan",
```

Comment:
Revert?
Could you revert all irrelevant changes, please? 😄

Reply:
Reverted now. Sorry about the reformatting changes.